lib/Extractor/Candidates.cpp

```diff
@@ -35,11 +35,16 @@
 #include <sstream>
 #include <unordered_set>
 #include <tuple>
+#include "llvm/Analysis/ValueTracking.h"
 
 static llvm::cl::opt<bool> ExploitBPCs(
     "souper-exploit-blockpcs",
     llvm::cl::desc("Exploit block path conditions (default=true)"),
     llvm::cl::init(true));
+static llvm::cl::opt<bool> HarvestKnownBits(
+    "souper-known-bits-analysis",
+    llvm::cl::desc("Perform known bits analysis (default = false)"),
```

Comment:
Sure, updating it in sclang.in too.

On Thu, Jul 16, 2015 at 12:06 PM, John Regehr <notifications@github.com>
wrote:

> In lib/Extractor/Candidates.cpp
> <https://github.com/google/souper/pull/195#discussion_r34818488>:
>
> >
> >  static llvm::cl::opt<bool> ExploitBPCs(
> >      "souper-exploit-blockpcs",
> >      llvm::cl::desc("Exploit block path conditions (default=true)"),
> >      llvm::cl::init(true));
> > +static llvm::cl::opt<bool> HarvestKnownBits(
> > +    "souper-known-bits-analysis",
> > +    llvm::cl::desc("Perform known bits analysis (default = false)"),
>
> please make the variable name and the command line option match up a bit
> better. so for example if the variable is HarvestKnownBits then make the
> option souper-harvest-known-bits
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/souper/pull/195/files#r34818488>.
>



-- 
Thanks and Regards,
Jubi Taneja